src/Aspire.Hosting.Redis/RedisBuilderExtensions.cs

```diff
@@ -55,22 +55,34 @@ public static IResourceBuilder<RedisResource> WithRedisCommander(this IResourceB
     }
 
     /// <summary>
-    /// Adds a named volume for the data folder to a Redis container resource.
+    /// Adds a named volume for the data folder to a Redis container resource and enables Redis persistence.
     /// </summary>
     /// <param name="builder">The resource builder.</param>
     /// <param name="name">The name of the volume. Defaults to an auto-generated name based on the resource name. </param>
     /// <param name="isReadOnly">A flag that indicates if this is a read-only volume.</param>
     /// <returns>The <see cref="IResourceBuilder{T}"/>.</returns>
     public static IResourceBuilder<RedisResource> WithDataVolume(this IResourceBuilder<RedisResource> builder, string? name = null, bool isReadOnly = false)
```

Comment:
Good question. I remember thinking about this and deciding to leave it there but now I can't remember why. I'll do some more testing today and verify the behavior.

Reply:
One could use the same volume between two different Redis resources, with one having only read access. I updated the methods so that if `isReadOnly == true` then Redis persistence is not configured, as the snapshot attempt will fail obviously if the directory is read only. I tried it out locally and it works well, with data from the volume being loaded but new operations not being persisted.